pkg/kubelet/nodestatus/setters.go

```diff
@@ -481,20 +483,49 @@ func GoRuntime() Setter {
 }
 
 // RuntimeClasses returns a Setter that sets RuntimeClasses on the node.
-func RuntimeClasses(fn func() []kubecontainer.RuntimeHandler) Setter {
+func RuntimeClasses(
+	rcManager *runtimeclass.Manager,
+	handlersFunc func() []kubecontainer.RuntimeHandler,
+) Setter {
 	return func(ctx context.Context, node *v1.Node) error {
-		if !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) {
+		if rcManager == nil || !utilfeature.DefaultFeatureGate.Enabled(features.RecursiveReadOnlyMounts) {
 			return nil
 		}
-		handlers := fn()
-		node.Status.RuntimeClasses = make([]v1.NodeRuntimeClass, len(handlers))
-		for i, h := range handlers {
-			node.Status.RuntimeClasses[i] = v1.NodeRuntimeClass{
-				Name: h.Name,
+		handlers := handlersFunc()
+		handlersMap := make(map[string]kubecontainer.RuntimeHandler, len(handlers))
+		for _, h := range handlers {
+			handlersMap[h.Name] = h
+		}
+
+		classes, err := rcManager.Lister().List(labels.Everything())
+		if err != nil {
+			return fmt.Errorf("failed to get the list of the runtime classes: %w", err)
+		}
+		node.Status.RuntimeClasses = []v1.NodeRuntimeClass{}
```

Comment:
How is this API consumed? Or is it informational only? I wonder if the API should just be a list of RuntimeHandlers instead?